students/ginafs/session05/mailroom3.py

```diff
@@ -0,0 +1,222 @@
+    print()
+    print()
+    return True
+
+
+def letters_to_donors():
+    '''Uses the donations dictionary to create Thank you letter.txt files for each
+    donor in donations dict. Builds a file_name from the donor's name and timestamp.
+    Creates modified letters for donors with a single donation and for donors
+    with multiple donations '''
+
+    print("Printing letters to donors")
+    print()
+    for key in DONATIONS:
+        #form file name from key
+        file_name = str(key).replace(' ', '_')
+
+        #get time stamp
+        timestamp = str(datetime.datetime.now()).split('.')[0]
+
+        #format the timestamp into an valid and readable string
+        timestamp = str(timestamp).replace(' ', '_')
+        timestamp = str(timestamp).replace(':', '.')
+        timestamp = str(timestamp).replace('-', '.')
+        file_name += '.' + timestamp
+
+        #get the number of donations for the donor and the total donation amount
+        num_donations = len(DONATIONS[key])
+        total_amount = sum(DONATIONS[key])
+
+        #Build the string to write into the file. Use tuples to send in values
+        out_str = ""
+        if num_donations > 1:
+            out_str = MULTIPLE_DONATION_LETTER
+            out_tuple = (key, int(num_donations), float(total_amount))
+        else:
+            #donor with single donation does not need number of donations
+            out_str = SINGLE_DONATION_LETTER
+            out_tuple = (key, float(total_amount))
+
+        #create file and write into it
+        with open(file_name, 'w') as outfile:
+            outfile.write(out_str.format(*out_tuple))
+            outfile.close()
+    return True
+
+
+def exit_program():
+    '''Exit the program with a message'''
+    print("\nGoodbye! Thank you for using the Donor Management Program")
+    return False
+
+
+def print_border():
+    ''' prints border '''
+    print("-" * 74)
+
+def print_heading():
+    '''prints headings
+    defining column names'''
+    columns = [
+        "{:25}".format("Donor Name"),
+        "{:14}".format("Total Given"),
+        "{:12}".format("Num Gifts"),
+        "{:14}".format("Average Gift")
+    ]
+    # build heading string
+    heading_str = EDGE_STR
+    for i, heading in enumerate(columns, 1):
+        heading_str += heading + EDGE_STR
+    # remove trailing blank
+    print(heading_str[:-1])
+
+def print_donor_report():
+    '''prints donor report'''
+    print_border()
+    print_heading()
+    print_border()
+
+    for key in DONATIONS:
+        line_str = EDGE_STR
+        # Donor Name
+        line_str += "{:25}".format(key) + EDGE_STR
+        # Total Given
+        sum_gifts = sum(DONATIONS[key])
+        line_str += "{: 14.2f}".format(sum_gifts) + EDGE_STR
+        # Num Gifts
+        num_gifts = len(DONATIONS[key])
+        line_str += "{: 12d}".format(num_gifts) + EDGE_STR
+        # Average
+        avg = sum_gifts/num_gifts
+        line_str += "{: 14.2f}".format(avg) + EDGE_STR
```

Comment:
The comments strings you have here don't add anything as they are just copies of the variable names.  What your doing is simple enough to read that you probably don't need them.